### ISSUE TITLE
InertialForce should not set use_displaced_mesh to true by default

### ISSUE THREAD

**Issue by maxnezdyur** [idaholab/moose#28174]
### Motivation

In `InertialForce` we have params.set<bool>("use_displaced_mesh") = true;.
This technically correct if we calculate the density correctly. 
If we set it to false then it would always be correct with just using a constant density.
$\int_{\Omega_0} \rho_0 a d\Omega_0 =  \int_{\Omega} \rho  a d\Omega  \neq  \int_{\Omega} \rho_0  a d\Omega$ if the volume is change during deformation. Many of the tests using `InertialForce` just use a constant generic material for the density so they are doing it incorrectly by default. 

### Design

Get rid of params.set<bool>("use_displaced_mesh") = true;

### Impact

Physically correct. As far as I can see, I don't see any test with InertialForce that calculates the density correctly.

**Comment by maxnezdyur** [comment]
@ddm42 Can you try to fix this

**Comment by ddm42** [comment]
i guess

**Comment by lindsayad** [comment]
I don't know the physics, but isn't this like Force = mass * acceleration. Isn't the mass equal to the density integrated over the volume, and if the volume is changing via deformation, then this should be taken into account?

**Comment by maxnezdyur** [comment]
The mass of an element should not change no matter what deformation (conservation of mass). If an element decreases in volume by a half and the density is constant, then the mass is cut in half (mass = (1/2)*volume * density). In this case density should actually double. We actually have a density object https://mooseframework.inl.gov/source/materials/Density.html that does this but since it is in the `Misc` folder it is only combined examples and none with solid dynamics. They use it for thermal probems.

**Comment by ddm42** [comment]
The issue seems to be that when people specify density in inertialforce kernel without using the Density object, they would assume that the density will change as the volume changes for each element. However, if they didn't use a Density material object, the density does not update as it deforms, and therefore using the Eulerian formulation that requires an up-to-date density will give wrong answers. However, if we default the use_displaced_mesh to false, then the math will work out. This is mainly for safety in case someone forgets to use a Density object.

**Comment by ddm42** [comment]
@maxnezdyur Here's a question: If someone uses a Density object, but they want to use the reference configuration (Lagrangian formulation), will the updated density be used in IntertialForce? As far as I can tell, it seems that would be wrong.

**Comment by maxnezdyur** [comment]
> @maxnezdyur Here's a question: If someone uses a Density object, but they want to use the reference configuration (Lagrangian formulation), will the updated density be used in InertialForce? As far as I can tell, InertialForce would use the updated density, and that would be wrong.

Yes I think so. In a perfect world, we would redo the density material so that is we set `use_displaced_mesh` to true it gives the correct density and if we set it to false it just gives the reference density. In this case the user would have more control.

**Comment by ddm42** [comment]
Yes that would solve everything. And also maybe stop allowing people to specify a Density as a "number" parameter in the kernel.

**Comment by maxnezdyur** [comment]
Or just error out if `use_displaced_mesh=true` and they specify a density number. The density "number" works if we have `use_displaced_mesh=false`
